arbnode/resourcemanager/resource_management_test.go

```diff
@@ -37,42 +36,58 @@ total_rss 1016209408
 total_inactive_file %d
 total_active_file 321544192
 `, inactive)
-	if err != nil {
-		return err
-	}
-	return nil
+	return err
 }
 
-func TestCgroupsv1MemoryLimit(t *testing.T) {
-	cgroupDir := t.TempDir()
-	c := newCgroupsV1MemoryLimitChecker(cgroupDir, 95)
-	_, err := c.isLimitExceeded()
-	if err == nil {
-		t.Error("Should fail open if can't read files")
+func makeCgroupsTestDir(cgroupDir string) cgroupsMemoryFiles {
+	return cgroupsMemoryFiles{
+		limitFile:  cgroupDir + "/memory.limit_in_bytes",
+		usageFile:  cgroupDir + "/memory.usage_in_bytes",
+		statsFile:  cgroupDir + "/memory.stat",
+		inactiveRe: regexp.MustCompile(`total_inactive_file (\d+)`),
 	}
+}
 
-	err = updateFakeCgroupv1Files(c, 1000, 1000, 51)
-	if err != nil {
-		t.Error(err)
-	}
-	exceeded, err := c.isLimitExceeded()
-	if err != nil {
-		t.Error(err)
-	}
-	if exceeded {
-		t.Error("Expected under limit")
+func TestCgroupsFailIfCantOpen(t *testing.T) {
+	testFiles := makeCgroupsTestDir(t.TempDir())
+	c := newCgroupsMemoryLimitChecker(testFiles, 95)
+	var err error
+	if _, err = c.isLimitExceeded(); err == nil {
```

Comment:
nit: same here, you don't need to define err before, you can just do

if _, err := ... ; err == nil {
}

It's more concise and limits the scope of err variable (less error prone in general that way for consequent error checks)

Reply:
Fixed.